File: spring-cloud-square-demo/consumer-retrofit-okhttp/src/main/java/com/livk/consumer/RemoteController.java
package com.livk.consumer;

import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpEntity;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.IOException;

/**
 * <p>
 * RemoteController
 * </p>
 *
 * @author livk
 * @date 2022/4/13
 */
@RestController
@RequiredArgsConstructor
public class RemoteController {

	private final RemoteService remoteService;

	@GetMapping("/remote/instance")
	public HttpEntity<String> remoteInstance() throws IOException {
		return ResponseEntity.ok(remoteService.instance().execute().body());
	}

}

File: spring-cloud-square/consumer-retrofit-okhttp/src/main/kotlin/com/livk/consumer/RemoteController.kt
package com.livk.consumer

import org.springframework.http.HttpEntity
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author livk
 */
@RestController
class RemoteController(private val remoteService: RemoteService) {
    @GetMapping("/remote/instance")
    fun remoteInstance(): HttpEntity<String> = ResponseEntity.ok(remoteService.instance().execute().body())
}
